blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java

```diff
@@ -115,6 +117,7 @@ public BluePipelineNodeContainer getNodes() {
     }
 
     @Override
+    @Navigable
```

Comment:
That is to only add links for steps and nodes on things that can actually have it. Freestyle e.g. implements the interface but will never have steps and nodes. Hence moving it to the implementation that actually can provide it